clang/test/SemaSYCL/sycl-templ-addr-space.cpp

```diff
@@ -0,0 +1,54 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -triple spir64 -verify %s
+
+// Test that an error is issued instead of an assertion for this test case.
+
+template <int b>
+struct c {
```

Comment:
Could you please try to make identifiers in this test a bit more memorable and meaningful? I think it will help to understand this test.

Reply:
The test case was reduced from a very large test case in #2763 using creduce (and then manually); that tool obfuscated the names.   I will try to give them better names, but I am not sure how much it will help.

Reply:
@Fznamznon, I have attempted to give the identifiers some meaningful names.

Reply:
Thank you!